common/util/autoware_point_types/include/autoware_point_types/types.hpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2021 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTOWARE_POINT_TYPES__AUTOWARE_POINT_TYPES_HPP_
+#define AUTOWARE_POINT_TYPES__AUTOWARE_POINT_TYPES_HPP_
+
+#include <point_cloud_msg_wrapper/point_cloud_msg_wrapper.hpp>
+
+#include <cmath>
+#include <limits>
+#include <tuple>
+
+namespace
+{
+template <class T>
+bool approximate_equal(const T a, const T b)
+{
+  const T e = std::numeric_limits<T>::epsilon();
+  const auto a_abs = std::fabs(a);
+  const auto b_abs = std::fabs(b);
+  return std::fabs(a) - std::fabs(b) <= (a_abs < b_abs ? b_abs : a_abs) * e;
```

Comment:
Did you write this function by yourself? I guess there are some bugs.

Reply:
So I recommend adding tests.

Reply:
Sorry, I got mistake. As you guess `std::fabs(a-b)` is right.
I'll add test code.

Reply:
I fixed in https://github.com/tier4/AutowareArchitectureProposal.iv/pull/784/commits/47d38e2eb8f33ca8d7bb6947cca3c455d80e6b8c and add test https://github.com/tier4/AutowareArchitectureProposal.iv/pull/784/commits/d928656866c460343577a6f17466c67c5bb0d64e